tests/pfcwd/test_pfcwd_warm_reboot.py

```diff
@@ -502,7 +502,7 @@ def pfcwd_wb_helper(self, request, testcase_actions, setup_pfc_test, fanout_grap
 
                     self.traffic_inst = SendVerifyTraffic(self.ptf, dut_facts['router_mac'], self.pfc_wd, queue)
                     self.run_test(port, queue, detect=(bitmask & 1),
-                                  storm_start=not t_idx or storm_deferred or storm_restored,
+                                  storm_start=('detect' in test_action) or storm_deferred or storm_restored,
```

Comment:
Can you please elaborate on the actual issue seen? if multiple ports are selected for the test, we want the fanout to send pause frames only for the 1st port. For the remaining ports, we are using a fake storm (look at set_storm_status() where we trigger a storm condition by setting some value for that port and queue in counters db as opposed to using the fanout to generate pause frames).

The change proposed above will cause pfc pause frames to generated by the fanout for all test ports.

Reply:
From my perspective there are should not be difference between test flow on one port or on few ports.
Few words about primary issue:
Test use some kind of randomization. Depending on what day it is the test uses today's  date (15 for today for example) as the number for the test and do some simple math. For most days of month number of ports for test will be only one and test will pass successfully.
But  on some days test will take 2 ports or even 3 
You can get familiar with randomization logic there: 
https://github.com/Azure/sonic-mgmt/blob/master/tests/pfcwd/files/pfcwd_helper.py#L236
In that case logic of test wasn't able to handle properly testing on few ports.
**self.run_test** called with params that later cause failing of test.

Reply:
@ayurkiv-nvda, there is a diff in the storm generation mechanism for the 1st port and the rest of the ports. if the test selects 'n' number of ports (where n >=1), for the 1st port, actual storm is sent from the fanout. for the rest of the ports, storm is faked. 
for your comment "In that case logic of test wasn't able to handle properly testing on few ports.
self.run_test called with params that later cause failing of test.", can you please elaborate more on what is the actual error message seen?
@lolyu , have you seen a similar issue as that stated in the PR?

Reply:
@neethajohn, no, I haven't met such an issue.
@ayurkiv-nvda, could you please elaborate why the fake storm with multiple ports leads to such an issue?